environments/skeleton/{{cookiecutter.environment}}/terraform/compute/nodes.tf

```diff
@@ -1,3 +1,11 @@
+data "external" "nodes" {
+  program = ["bash", "-c", <<-EOT
+    openstack baremetal node list --limit 0 -f json 2>/dev/null | \
+    jq -r 'try map( { (.Name|tostring): .UUID } ) | add catch {}' || echo '{}'
```

Comment:
Is this so it behaves well when Ironic is not enabled?

Reply:
Yes, or e.g. you're not admin/don't have policy amendments and deploying a VM cluster hence can't/don't need to query BM nodes at all

Reply:
Writing that out made me think though - it'd be better if it worked so IF `match_ironic_node` is true, this command has to work, and if you don't this command is actually a no-op.

Reply:
actually we can't really gate the command like that, given `match_ironic_node` could be true on some nodes and not others.

Reply:
Turns out setting a default of `nova` for the AZ doesn't work, b/c if you don't have admin/right policy, you can't set this at all...